### PR TITLE
Update workflows to new model

### PR DESCRIPTION
**Pull request by vgrassia**
This PR is for updating our existing pipelines to the Build/Test/Release model.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -29,312 +22,147 @@ jobs:
...
-        uses: actions/upload-release-asset@e8f9f06c4b078e705bd2ea027f0926603fc9b4d5  # 1.0.2 - Archived
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.setup.outputs.release_upload_url }}
-          asset_name: bw-snap-sha256-${{ env._PACKAGE_VERSION }}.txt
-          asset_path: ./dist/snap/bw-snap-sha256-${{ env._PACKAGE_VERSION }}.txt
-          asset_content_type: plain/text
+      - name: Publish NPM
+        if: false
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:167]: Looks like there is some testing code still in here.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,312 +22,147 @@ jobs:
...
+          workflow_conclusion: success
+          branch: rc
+          artifacts: bitwarden-cli.${{ env._PKG_VERSION }}.nupkg
 
-      - name: Setup Snap Package
-        run: |
-          cp -r stores/snap/* -t dist/snap
-          sed -i s/__version__/${{ env._PACKAGE_VERSION }}/g dist/snap/snapcraft.yaml
+      - name: Push to Chocolatey
+        if: false
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:139]: Looks like there is some testing code still in here.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -29,312 +22,147 @@ jobs:
...
-        uses: actions/upload-release-asset@e8f9f06c4b078e705bd2ea027f0926603fc9b4d5  # 1.0.2 - Archived
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.setup.outputs.release_upload_url }}
-          asset_name: bitwarden-cli.${{ env.PACKAGE_VERSION }}.nupkg
-          asset_path: ./dist/chocolatey/bitwarden-cli.${{ env.PACKAGE_VERSION }}.nupkg
-          asset_content_type: application
+      - name: Publish Snap & logout
+        if: false
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:107]: Looks like there is some testing code still in here.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -29,312 +22,147 @@ jobs:
...
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ needs.setup.outputs.release_upload_url }}
-          asset_name: bw-snap-sha256-${{ env._PACKAGE_VERSION }}.txt
-          asset_path: ./dist/snap/bw-snap-sha256-${{ env._PACKAGE_VERSION }}.txt
-          asset_content_type: plain/text
+      - name: Publish NPM
+        if: false
+        run: npm run publish:npm
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:168]: This is running an `npm run build:prod`. I wondering if we can build this in the previous workflow, zip it, and unzip it to the right directory here before running the `npm publish --access public`

**Review by joseph-flinn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (vgrassia) [reply .github/workflows/release.yml:167]: Removed.

File: .github/workflows/release.yml (hunk 2 above)
Comment (vgrassia) [reply .github/workflows/release.yml:139]: Removed.

File: .github/workflows/release.yml (hunk 3 above)
Comment (vgrassia) [reply .github/workflows/release.yml:107]: Removed.

File: .github/workflows/release.yml (hunk 4 above)
Comment (vgrassia) [reply .github/workflows/release.yml]: I added an artifact that is the whole `build` folder which should be all that's needed for the `npm publish` command.  Let me know what you think.

**Review by joseph-flinn — APPROVED** [review]

File: .github/workflows/release.yml (hunk 4 above)
Comment (joseph-flinn) [reply .github/workflows/release.yml]: I like it!

**Pull request merged**